test/api_compatibility.js

```diff
@@ -1,46 +1,48 @@
 // For the convenience of unit testing, add these to the global namespace
 var _      = require('underscore');
+var assert = require('chai').assert;
 var expect = require('chai').expect;
+var hasSpanFunctions = require('./api_compatibility_helpers.js').hasSpanFunctions;
+var hasTracerFunctions = require('./api_compatibility_helpers.js').hasTracerFunctions;
+var hasSpanContextFunctions = require('./api_compatibility_helpers.js').hasSpanContextFunctions;
 
 // Unit testing is done against the debug version of the library as it has
 // additional conformance checks that are optimized out of the production
 // library.  Again, globals are used purely for convenience.
 var opentracing = require('../debug.js');
 
-module.exports = function apiCompatibilityChecks(_createTracer) {
+// TODO(oibe) update documentation to address carrier factory map
+module.exports = function apiCompatibilityChecks(_createTracer, carrierFormatFactoryMap, checkBaggageValues) {
```

Comment:
What type of object the carrier is depends on the format, nothing else. What that object is populated with is implementation dependent, but you do not need to know what it's populated with if your test start with empty one, do inject, then do extract from the result of inject. For example: https://github.com/uber/jaeger-client-go/blob/master/propagation_test.go#L24